packages/frameworks-solid-start/src/client.tsx

```diff
@@ -0,0 +1,322 @@
+import type {
+  LiteralUnion,
+  SignInOptions,
+  SignInAuthorizationParams,
+  SignOutParams,
+} from "@auth/core/types"
+import type {
+  BuiltInProviderType,
+  RedirectableProviderType,
+} from "@auth/core/providers"
+import { Session } from "@auth/core/types"
+import {
+  Accessor,
+  createContext,
+  createEffect,
+  createMemo,
+  createSignal,
+  onCleanup,
+  useContext,
+} from "solid-js"
+import {
+  UseSessionOptions,
+  AuthClientConfig,
+  SessionProviderProps,
+} from "./types"
+import { now, objectIsSession, parseUrl } from "./utils"
+
+const __NEXTAUTH: AuthClientConfig = {
+  baseUrl: parseUrl(process.env.AUTH_URL ?? process.env.VERCEL_URL).origin,
+  basePath: parseUrl(process.env.AUTH_URL).path,
+  baseUrlServer: parseUrl(
+    process.env.AUTH_URL_INTERNAL ??
+      process.env.AUTH_URL ??
+      process.env.VERCEL_URL
+  ).origin,
+  basePathServer: parseUrl(
+    process.env.AUTH_URL_INTERNAL ?? process.env.AUTH_URL
+  ).path,
+  _lastSync: 0,
+  _session: undefined,
+  _getSession: () => {},
+}
+
+export type SessionContextInner<R extends boolean = false> = R extends true
+  ?
+      | { data: Session; status: "authenticated" }
+      | { data: null; status: "loading" }
+  :
+      | { data: Session; status: "authenticated" }
+      | { data: null; status: "unauthenticated" | "loading" }
+export type SessionContextValue<R extends boolean = false> = Accessor<
+  SessionContextInner<R>
+>
+
+export const SessionContext = createContext<SessionContextValue | undefined>(
+  undefined
+)
+
+/**
+ * React Hook that gives you access
+ * to the logged in user's session data.
+ *
+ * [Documentation](https://next-auth.js.org/getting-started/client#usesession)
+ */
+
+export function createSession<R extends boolean>(
+  options?: UseSessionOptions<R>
+): Accessor<SessionContextInner<R>> {
+  // @ts-expect-error Satisfy TS if branch on line below
+  const value: SessionContextValue<R> = useContext(SessionContext)
+  if (!value && (import.meta as any).env.DEV) {
+    throw new Error(
+      "[solid-auth]: `useSession` must be wrapped in a <SessionProvider />"
+    )
+  }
+
+  const { required, onUnauthenticated } = options ?? {}
+
+  const requiredAndNotLoading = () =>
+    required && value().status === "unauthenticated"
+
+  createEffect(() => {
+    if (requiredAndNotLoading()) {
+      const url = `/api/auth/signin?${new URLSearchParams({
+        error: "SessionRequired",
+        callbackUrl: window.location.href,
+      })}`
+      if (onUnauthenticated) onUnauthenticated()
+      else window.location.href = url
+    }
+  })
+
+  if (requiredAndNotLoading()) {
+    return () => ({ data: value().data, status: "loading" }) as any
+  }
+
+  return value
+}
+
+/**
+ * Provider to wrap the app in to make session data available globally.
+ * Can also be used to throttle the number of requests to the endpoint
+ * `/api/auth/session`.
+ *
+ * [Documentation](https://next-auth.js.org/getting-started/client#sessionprovider)
```

Comment:
Comments should be fixed, this is going to be a dead link. (Likely not need a link anymore as the docs page is generated from the same source)